apps/web/src/pages/templates/components/ChannelStepEditor.tsx

```diff
@@ -1,81 +1,48 @@
-import { useFormContext } from 'react-hook-form';
-import { ChannelTypeEnum, providers, StepTypeEnum } from '@novu/shared';
+import { useParams } from 'react-router-dom';
+import { StepTypeEnum } from '@novu/shared';
+
 import { EmailMessagesCards } from './email-editor/EmailMessagesCards';
 import { TemplateInAppEditor } from './in-app-editor/TemplateInAppEditor';
 import { TemplateSMSEditor } from './TemplateSMSEditor';
-import type { IForm } from './formTypes';
 import { TemplatePushEditor } from './TemplatePushEditor';
 import { TemplateChatEditor } from './chat-editor/TemplateChatEditor';
-import {
-  useActiveIntegrations,
-  useEnvController,
-  useGetPrimaryIntegration,
-  useHasActiveIntegrations,
-} from '../../../hooks';
-import { useNavigate, useParams } from 'react-router-dom';
 import { SubPageWrapper } from './SubPageWrapper';
 import { DigestMetadata } from '../workflow/DigestMetadata';
 import { DelayMetadata } from '../workflow/DelayMetadata';
 import { colors } from '../../../design-system';
-import React, { useEffect, useMemo } from 'react';
-import { useBasePath } from '../hooks/useBasePath';
 import { StepName } from './StepName';
 import { DeleteStepRow } from './DeleteStepRow';
 import { TranslateProductLead } from './TranslateProductLead';
-import { DisplayPrimaryProviderIcon } from '../workflow/DisplayPrimaryProviderIcon';
-import { getChannel } from '../../../utils/channels';
-import { CONTEXT_PATH } from '../../../config';
-import { useMantineColorScheme } from '@mantine/core';
-import styled from '@emotion/styled';
+import { useStepIndex } from '../hooks/useStepIndex';
+import { useNavigateFromEditor } from '../hooks/useNavigateFromEditor';
 
 export const ChannelStepEditor = () => {
-  const { readonly } = useEnvController();
-
-  const { channel, stepUuid = '' } = useParams<{
+  const { channel } = useParams<{
     channel: StepTypeEnum | undefined;
-    stepUuid: string;
   }>();
-  const { integrations } = useActiveIntegrations();
-  const {
-    control,
-    formState: { errors },
-    watch,
-  } = useFormContext<IForm>();
-  const steps = watch('steps');
-
-  const index = useMemo(
-    () => steps.findIndex((message) => message.template.type === channel && message.uuid === stepUuid),
-    [channel, stepUuid, steps]
-  );
-
-  const navigate = useNavigate();
-  const basePath = useBasePath();
+  const { stepIndex, variantIndex } = useStepIndex();
+  const key = `${stepIndex}_${variantIndex}`;
```

Comment:
this key is used to re-render the editors, for example, when the URL changes... in the case when picking the different SMS nodes from the workflow, we need to re-render the SMS editor